assets/js/modules/adsense/util/site-stats-data.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Site stats utlities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Gets data for a Google Chart from an adesnse report.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} current        A report object for the current period.
+ * @param {Object} previous       A report object for the previous period.
+ * @param {string} label          Metric label.
+ * @param {number} selectedColumn Selected column index.
+ * @return {Array.<Array.<number|string>>} Data array.
+ */
+export function getSiteStatsDataForGoogleChart( current, previous, label, selectedColumn ) {
+	const dataMap = [
+		[
+			{ type: 'date', label: __( 'Day', 'google-site-kit' ) },
+			{ type: 'number', label },
+			{ type: 'number', label: __( 'Previous period', 'google-site-kit' ) },
+		],
+	];
+
+	const stringToDate = ( dateString ) => new Date( `${ dateString } 00:00:00` );
+	const findRowByDate = ( searchDate ) => ( ( [ rowDate ] ) => searchDate.getTime() === stringToDate( rowDate ).getTime() );
+
+	const currentDate = stringToDate( current.startDate );
+	const previousDate = stringToDate( previous.startDate );
+
+	const ends = stringToDate( current.endDate );
+
+	while ( +currentDate <= +ends ) {
+		dataMap.push( [
+			new Date( currentDate ), // Copy the current date.
+			parseFloat( ( current?.rows || [] ).find( findRowByDate( currentDate ) )?.[ selectedColumn ] || 0 ),
+			parseFloat( ( previous?.rows || [] ).find( findRowByDate( previousDate ) )?.[ selectedColumn ] || 0 ),
+		] );
+
+		currentDate.setDate( currentDate.getDate() + 1 );
+		previousDate.setDate( previousDate.getDate() + 1 );
+	}
+
+	return dataMap;
+}
```

Comment:
Can you update the formatting in this function to result in similar tooltips like the Analytics graph on the Analytics page has? See image below:

![Screenshot 2020-12-08 at 10 01 11](https://user-images.githubusercontent.com/3531426/101527682-2f0e7200-3943-11eb-8c2c-3fc89940ce8a.png)

Compared to current state here:
![Screenshot 2020-12-08 at 10 00 55](https://user-images.githubusercontent.com/3531426/101527838-5f561080-3943-11eb-9413-e32f67c5f1be.png)

In other words, there should be a percentage indicator instead of the second absolute number, both dates should be displayed, and the overall tooltip UI should look similar to the Analytics tooltips.

Can you also check the number formatting to be according to the metrics? The earnings and RPM should have a currency included, and the CTR should be a percentage value (that number in particular right now is furthermore incorrect because it's divided by 100, so the current AdSense tooltips display e.g. 0.01 instead of 1%).

Reply:
Updated.